common/functions/src/scalars/strings/repeat.rs

```diff
@@ -0,0 +1,234 @@
+use crate::scalars::Function;
+
+macro_rules! const_array {
+    ($input_string:expr, $times:expr, $input_rows:expr) => {{
+        let mut string_builder = StringArrayBuilder::with_capacity($input_rows);
+        for n in $times {
+            let value = n.and_then(|n| {
+                if *n >= 0 {
+                    Some($input_string.repeat(*n as usize))
+                } else {
+                    None
+                }
+            });
+            string_builder.append_option(value);
+        }
+        Ok(string_builder.finish().into())
+    }};
+}
+
+macro_rules! const_array_unsigned {
+    ($input_string:expr, $times:expr, $input_rows:expr) => {{
+        let mut string_builder = StringArrayBuilder::with_capacity($input_rows);
+        for n in $times {
+            string_builder.append_option(n.map(|n| $input_string.repeat(*n as usize)));
+        }
+        Ok(string_builder.finish().into())
+    }};
+}
+
+macro_rules! array_array {
+    ($input_array:expr, $times:expr, $input_rows:expr) => {{
+        let mut string_builder = StringArrayBuilder::with_capacity($input_rows);
+        for (value, n) in $input_array.into_iter().zip($times) {
+            string_builder.append_option(value.zip(n).and_then(|(value, n)| {
+                if *n >= 0 {
+                    Some(value.repeat(*n as usize))
+                } else {
+                    None
+                }
+            }));
+        }
+        Ok(string_builder.finish().into())
+    }};
+}
+
+macro_rules! array_array_unsigned {
+    ($input_array:expr, $times:expr, $input_rows:expr) => {{
+        let mut string_builder = StringArrayBuilder::with_capacity($input_rows);
+        for (value, n) in $input_array.into_iter().zip($times) {
+            string_builder.append_option(value.zip(n).map(|(value, n)| value.repeat(*n as usize)));
+        }
+        Ok(string_builder.finish().into())
+    }};
+}
+
+#[derive(Clone)]
+pub struct RepeatFunction {
+    _display_name: String,
+}
+
+impl RepeatFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(RepeatFunction {
+            _display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for RepeatFunction {
+    fn name(&self) -> &str {
+        "repeat"
+    }
+
+    fn num_arguments(&self) -> usize {
+        2
+    }
+
+    fn variadic_arguments(&self) -> Option<(usize, usize)> {
+        None
+    }
+
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if !matches!(args[0], DataType::String) {
+            return Err(ErrorCode::IllegalDataType(format!(
+                "Expected parameter 1 is string, but got {}",
+                args[0]
+            )));
+        }
+
+        if !args[1].is_integer() {
```

Comment:
let's just accept `args[1]` as  `unsigned integer` | `String` | `Null`.

Then we can cast column[1] to UInt64Column to reduce the matches.

Reply:
Cast is not expensive because it's vectorized.

Reply:
> Then we can cast column[1] to UInt64Column to reduce the matches.

Will this cause unnecessary memory allocation?

Reply:
Yes, but I think it's ok. The old column memory will be dropped immediately.

In most case, the second column is a `DataColumn::Constant`, so we can use `let times = columns[1].to_minimal_array().cast_with_type(&DataType::UInt64)`.

Reply:
Since we already wrote the match macro, it's ok to not change that :)

Let's add String|Null support!